src/pykeen/triples/triples_factory.py

```diff
@@ -540,45 +651,84 @@ def get_most_frequent_relations(self, n: Union[int, float]) -> Set[str]:
         elif not isinstance(n, int):
             raise TypeError('n must be either an integer or a float')
 
-        counter = Counter(self.triples[:, 1])
-        return {
-            relation
-            for relation, _ in counter.most_common(n)
-        }
+        uniq, counts = self.mapped_triples[:, 1].unique(return_counts=True)
+        top_counts, top_ids = counts.topk(k=n, largest=True)
+        return set(uniq[top_ids].tolist())
 
-    def get_idx_for_entities(self, entities: Collection[str], invert: bool = False):
-        """Get np.array indices for triples with the given entities."""
-        entities = np.asanyarray(entities, dtype=self.triples.dtype)
-        return (
-            np.isin(self.triples[:, 0], entities, invert=invert)
-            & np.isin(self.triples[:, 2], entities, invert=invert)
+    def entities_to_ids(self, entities: Collection[Union[int, str]]) -> Collection[int]:
```

Comment:
Might be better to type as `Union[Collection[int], Collection[str]]`. The implementation is agnostic, but I think this gives a better idea what's going on from a documentation standpoint

Reply:
Same for all of the other functions using this one and the equivalent relation one

Reply:
The former one allows mixed types, so this would restrict the API. However, I am not sure whether there is a use case for providing mixed ones :sweat_smile: